libraries/rush-lib/src/cli/RushCommandLineAPI.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  CommandLineAction,
+  CommandLineParameter,
+  CommandLineParameterKind
+} from '@rushstack/ts-command-line';
+import { RushCommandLineParser } from './RushCommandLineParser';
+
+interface ICommandLineParameter {
+  /**
+   * The corresponding string representation of CliParameterKind
+   */
+  readonly kind: string;
+
+  /**
+   * The long name of the flag including double dashes, e.g. "--do-something"
+   */
+  readonly longName: string;
+
+  /**
+   * An optional short name for the flag including the dash, e.g. "-d"
+   */
+  readonly shortName?: string;
+
+  /**
+   * Documentation for the parameter that will be shown when invoking the tool with "--help"
+   */
+  readonly description: string;
+
+  /**
+   * If true, then an error occurs if the parameter was not included on the command-line.
+   */
+  readonly required?: boolean;
+}
+
+interface IRushCliJsonSpec {
+  actionName: string;
+  parameters: ICommandLineParameter[];
+}
+
+/**
+ * Information about the available CLI commands
+ *
+ * @beta
+ */
+export class RushCommandLineAPI {
```

Comment:
Our capitalization convention would be `RushCommandLineApi` (instead of `RushCommandLineAPI`).

However looking at **rush-lib.api.md** as a whole, the natural name for this API I'm thinking would be just `RushCommandLine`. That would cause confusion with the existing `RushXCommandLine` class whose purpose is somewhat unrelated; however that class is entirely internal, so we could rename it. We can also move the RushCommandLine definition under `src/api` instead of `src/cli` to further clarify this separation.

Reply:
Changed to RushCommandLine and moved to src/api